src/utils/config.ts

```diff
@@ -61,7 +61,7 @@ export function intVar<R>(varName: string, defaultValue?: R): number | R {
 // resolve number of workers based on number of CPUs assigned to pods or available CPUs
 export const getInstanceCount = (varName: string) => {
 	let instanceCount = intVar(varName, null);
-	if (instanceCount != null) {
+	if (instanceCount != null && instanceCount > 0) {
```

Comment:
Why is it set to 0?

Reply:
0 means autodetect

Reply:
Don't we already have unset for that @wrboyce ?

Reply:
Not really, `VPN_INSTANCE_COUNT` is required to be set or the app won't start.

Reply:
What requires it? And if it's required why are we treating it nullable here? Also what will happen for the openvpn processes if it's set to 0?

Reply:
> What requires it?

app.ts

> Also what will happen for the openvpn processes if it's set to 0?

one openvpn process is spawned by each worker regardless of this value.

Reply:
But app.ts also has handling for it not being set? Why do we have code to handle it not being set if we also require it to be set, it doesn't make sense imo

As for openvpn I realized what I actually meant was the haproxy config

Reply:
@Page- @wrboyce are we good to merge on this one?

Reply:
I'm not sure what you're asking about haproxy @Page-? Each worker will register itself as a backend worker with haproxy on startup.

Reply:
> But app.ts also has handling for it not being set?

I'd argue it has handling for it being set to an invalid value.

Reply:
What is the
```
{{- /* servers = getenv(VPN_INSTANCE_COUNT, default=getenv(NPROC) if production else 1) */ -}}
{{ $servers := (getenv "VPN_INSTANCE_COUNT" (or (and (eq (getenv "PRODUCTION_MODE") "true") (getenv "NPROC")) "1")) -}}
```
stuff for in the haproxy config @wrboyce ?

And yeah I think it's fine @ab77, at worst it'll be another PR for haproxy :P

Reply:
Hmm good point Page, that should actually fall back to `getenv(NPROC)` if `VPN_INSTANCE_COUNT` is <1.

@ab77 is this change actually required here? We set `VPN_INSTANCE_COUNT` to `NPROC` in `env.tmpl` if it isn't explicitly set or if it is set to zero.

```
VPN_INSTANCE_COUNT={{ if gt (getv "/resin/vpn/instances" "0") "0" }}{{ getv "/resin/vpn/instances" }}{{ else }}{{ if eq (getv "/resin/vpn/production") "true" }}{{getenv "NPROC"}}{{ else }}1{{ end }}{{ end }}
```

Reply:
For reference `NPROC` is set [here](https://github.com/balena-io/open-balena-vpn/blob/master/config/services/confd.service.d/confd.conf).